akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -79,36 +79,12 @@ trait PredefinedToResponseMarshallers extends LowPriorityToResponseMarshallerImp
         }
     })
 
-  implicit def fromEntityStreamingSupportAndByteStringMarshaller[T, M](implicit s: EntityStreamingSupport, m: ToByteStringMarshaller[T]): ToResponseMarshaller[Source[T, M]] = {
-    Marshaller[Source[T, M], HttpResponse] { implicit ec ⇒ source ⇒
-      FastFuture successful {
-        Marshalling.WithFixedContentType(s.contentType, () ⇒ {
-          val availableMarshallingsPerElement = source.mapAsync(1) { t ⇒ m(t)(ec) }
-
-          // TODO optimise such that we pick the optimal marshalling only once (headAndTail needed?)
-          // TODO, NOTE: this is somewhat duplicated from Marshal.scala it could be made DRYer
-          val bestMarshallingPerElement = availableMarshallingsPerElement mapConcat { marshallings ⇒
-            // pick the Marshalling that matches our EntityStreamingSupport
-            (s.contentType match {
-              case best @ (_: ContentType.Binary | _: ContentType.WithFixedCharset | _: ContentType.WithMissingCharset) ⇒
-                marshallings collectFirst { case Marshalling.WithFixedContentType(`best`, marshal) ⇒ marshal }
-
-              case best @ ContentType.WithCharset(bestMT, bestCS) ⇒
-                marshallings collectFirst {
-                  case Marshalling.WithFixedContentType(`best`, marshal) ⇒ marshal
-                  case Marshalling.WithOpenCharset(`bestMT`, marshal)    ⇒ () ⇒ marshal(bestCS)
-                }
-            }).toList
-          }
-          val marshalledElements: Source[ByteString, M] =
-            bestMarshallingPerElement.map(_.apply()) // marshal!
-              .via(s.framingRenderer)
+  @deprecated("This method exists only for the purpose of binary compatibility, it used to be implicit.", "10.1.0")
+  def fromEntityStreamingSupportAndByteStringMarshaller[T, M](s: EntityStreamingSupport, m: ToByteStringMarshaller[T]): ToResponseMarshaller[Source[T, M]] =
+    fromEntityStreamingSupportAndByteStringMarshaller(null, s, m)
```

Comment:
hm not enjoying passing nulls around

Reply:
Isn't this calling itself and wound cause a n infinite recursion..?

Reply:
so that's the classtag... configusing to pass the null... would it now blow up for existing users?

Reply:
I believe it won't; the method that uses the class tag, `NoStrictlyCompatibleElementMarshallingAvailableException`, can cope with a null class tag (because the same overloading with `null` was used when `fromEntityStreamingSupportAndEntityMarshaller` was changed.

But I agree it's unideal; I could change the private method to take an `Option[ClassTag[T]]`?

Reply:
On reflection, I'm not entirely sure using an option would be a good idea; since the sole user of the `ClassTag[T]`, `NoStrictlyCompatibleElementMarshallingAvailableException`, was written to explicitly handle a `null` (see commit https://github.com/akka/akka-http/commit/238e7fb553948978ab9d6cac6aa5f164b62de66d).

Happy to make any chance you'd prefer, though.

Reply:
The null is indeed a bit dirty but I guess acceptable as it's a 'feature' of `NoStrictlyCompatibleElementMarshallingAvailableException` anyway... might warrant a comment though? Do/can we have a test that checks this situation?

Reply:
I've added a test case (which also tests `NoStrictlyCompatibleElementMarshallingAvailableException` being thrown by `fromEntityStreamingSupportAndByteStringMarshaller`).